data/data.rb

```diff
@@ -0,0 +1,169 @@
+require_relative '../book'
+require_relative '../person'
+require_relative '../student'
+require_relative '../teacher'
+require_relative '../app'
+require 'json'
+
+def load_books
+  if File.exist?('./data/books.json')
+    file = File.open('./data/books.json')
+
+    # File.empty?(path)
+    if File.empty?('./data/books.json')
+      'If this is your first time using our app, kindly add some books first.'
+    else
+      books = JSON.parse(File.read('./data/books.json'))
+
+      books.each do |book|
+        book = Book.new(book['title'], book['author'])
+        @books << book
+      end
+    end
+    file.close
+  end
+end
+
+def save_book(title, author)
+  obj = {
+    title: title,
+    author: author
+  }
+
+  if File.exist?('./data/books.json')
+    file = File.open('./data/books.json')
+
+    if File.empty?('./data/books.json')
+      book = [obj]
+    else
+      book = JSON.parse(File.read('./data/books.json'))
+      book << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/books.json', 'w')
+    myfile.write(JSON.pretty_generate(book))
+    myfile.close
+  end
+end
```

Comment:
- [ ] I noticed that the books, people and rentals are saved immediately after they are created. Based on the requirement,  the data created when using the application should be saved to JSON files on exit from the application and not immediately after they are created. I would advise you to call the `save_rental`, `save_book`, `save_teacher` and `save_student` methods upon exit from the application. 👍